### ISSUE TITLE
Archlinux setup?

### ISSUE THREAD

**Issue by freshLobster** [Make-Magazine/PirateRadio#10]
I'm trying to get this working on ArchLinuxARM because my Pi's sd card is smaller and nothing is working quite right. The os is all good and dandy with python etc.. I tried using python 2.yadda and 3.yadda, neither work right. I edited a bit of the code (import configparser and the location of file pifm) and changed the defaults because the config file wouldn't load right in python2. In python 3 it wouldn't run at all. I finally got it to run, but it won't play music and essentially jams whatever frequency I use in the defaults, even commercial stations. Albeit this is pretty cool, its not what I was trying to do. What could be going wrong here? I've been working on this for awhile with no success. Have you been able to get this to run in arch? in the future could there possibly be a prepackaged archARM image like the Raspian one on the make site since it takes up less sd card space?

**Comment by ric96** [comment]
For some reason i get the samee issuee on raspbian try running it using sudo ./pifm

**Comment by zerotri** [comment]
The image I built for Make is actually based on Arch, as I'm a bit frustrated with Debian's packages being out of date, sometimes by even a few years, when I need the latest and greatest for some of my projects. I also didn't want to take the time cutting all of the DEs and such out of Raspbian for this.

I would like to make another image though, with a more streamlined system, and adding in some extra features such as NFS/SMB shares and broadcasting of these via zeroconf to make it easy to add music over the network. Really quite a few things I would like to add.

Using Python2 is a known issue. I've actually got some code in the 'unstable' branch for automatically picking the correct imports for the python version running that I'd like to test but haven't gotten the time yet; You're free to give that a whirl and let me know if it works.

It sounds like the pifm binary is running fine but there is an issue with piping sound to it. Do you have FFMPEG installed? If not, it would certainly cause this issue. FFMPEG is used to convert the audio files from whatever format they may be in to something usable by the pifm binary and pipes the audio data to it.

Please double check that FFMPEG is installed and let me know.

**Comment by freshLobster** [comment]
Yeah I have ffmpeg installed and I just decided to stick with python 2 because 3 had more weird errors, but so far its the same; signal broadcast, no config file load, no audio file played.

**Comment by zerotri** [comment]
Can you please create a gist with your version of the PirateRadio.py as well as a list of the locations of your PirateRadio.py, pifm and PirateRadio.config? If it is having issues loading still it is likely an issue with the locations.
